Enhance README.md files for panel plugins (#253)
grafana/grafana-plugin-examples: examples/panel-basic/README.md

# Basic Panel Plugin for Grafana

> This repository contains an example plugin for Grafana called Basic Panel Plugin. It serves as a foundational example for creating custom panel plugins within Grafana.

Reviewer: ```suggestion
Basic Panel Plugin is a foundational example for creating custom panel plugins within Grafana.
```

"This repository" feels odd when the user is currently in a subfolder? Similar concern for the rest of the overviews

## Overview

The Basic Panel Plugin demonstrates the fundamental structure and functionality required for developing custom panels in Grafana. It provides a starting point for developers looking to create their own custom visualizations or extend Grafana's capabilities. 

Panel plugins allow you to add new types of visualizations to your dashboard, such as maps, clocks, pie charts, lists, and more. Use panel plugins when you want to do things like visualize data returned by data source queries, navigate between dashboards, or control external systems (such as smart home devices).

> The definition for the Basic Panel Plugin is provided in [`/src/components/SimplePanel/SimplePanel.tsx`](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/panel-basic/src/components/SimplePanel/SimplePanel.tsx).

Reviewer: This doesn't make sense to me, the `SimplePanel.tsx` is just a component of the plugin, albeit the main one - but the panel options are part of it too and defined elsewhere in the `module.ts`. But it isn't the definition of the plugin. The `plugin.json` is arguably the closest to that. 

I'm not sure what we want to convey here 🤔

> The definition for the Basic Panel Plugin is provided in []`/src/components/SimplePanel/SimplePanel.tsx`](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/panel-basic/src/components/SimplePanel/SimplePanel.tsx).

Reviewer 2: ```suggestion
The definition for the Basic Panel Plugin is provided in [`/src/components/SimplePanel/SimplePanel.tsx`](https://github.com/grafana/grafana-plugin-examples/blob/main/examples/panel-basic/src/components/SimplePanel/SimplePanel.tsx).
```

> The plugin uses `TimeSeries` from the `grafana-ui` package to build a graph with the properties passed to the panel. The plugin also allows a tooltip to be shown when the user hovers over a visualization. Additionally, the plugin is set up to allow custom options such as a gradient mode selector and a list display mode to be configured from the Grafana sidebar.

Reviewer: This section might be more useful if it linked to snippets in the code
